python_modules/dagster/dagster/_core/asset_graph_view/serializable_entity_subset.py

```diff
@@ -80,6 +80,36 @@ def is_compatible_with_partitions_def(
         else:
             return partitions_def is None
 
+    def _check_incoming_value_is_compatible(self, value: EntitySubsetValue) -> None:
+        incoming_is_partitioned = not isinstance(value, bool)
+        if self.is_partitioned and not incoming_is_partitioned:
+            raise ValueError(
+                f"Cannot add {value} to subset. The types are incompatible. EntitySubset is partitioned"
+            )
+        if not self.is_partitioned and incoming_is_partitioned:
+            raise ValueError(
+                f"Cannot add {value} to subset. The types are incompatible. EntitySubset is not partitioned"
+            )
+
+    def add(self, value: EntitySubsetValue) -> "SerializableEntitySubset":
```

Comment:
this API feels a bit odd for non-partitioned assets when i use it. Maybe for non-partitioned assets you pass None instead of the boolean...